examples/eks/README.md

```diff
@@ -0,0 +1,226 @@
+
+> This tutorial was tested on a machine running Ubuntu 20.04 with a EKS cluster based on EC2 instances.
+
+## Starting and testing a deployment of Connaisseur
+
+### 1. Set up environment
+
+First off, we'll export the variables used in this tutorial to allow you to immediately use your own names/URLs instead of relying on our default names. Below, substitute your own values as appropriate:
+
+```bash
+REGISTRY_URL=docker.io
+REPOSITORY_NAME=testingconny
+IMAGE_NAME=testserver
+IMAGE_PATH=$(echo ${REGISTRY_URL}/${REPOSITORY_NAME}/${IMAGE_NAME})
+
+NOTARY_URL=notary.docker.io
+NOTARY_USER=<YOUR-NOTARY-USERNAME/DOCKER-HUB-ID>
+NOTARY_PASSWORD=<YOUR-NOTARY-PASSWORD>
+
+git clone https://github.com/sse-secure-systems/connaisseur.git
+cd connaisseur
+```
+
+### (Optional) 2. Prepare test images
+
+Go ahead and build both an unsigned and a signed image:
+
+```bash
+cd examples
+docker build -f Dockerfile.unsigned -t ${IMAGE_PATH}:unsigned .
+docker push ${IMAGE_PATH}:unsigned
+docker build -f Dockerfile.signed -t ${IMAGE_PATH}:signed .
+DOCKER_CONTENT_TRUST=1 docker push ${IMAGE_PATH}:signed
+cd -
+```
+
+During the second push, you will be asked to create a root key password and a repository key password (unless you already pushed signed images to the registry and repository, respectively).
+
+### 3. Configure Connaisseur
+
+Before you can finally set up Connaisseur, you will need to configure its connection to the notary server provided by Docker Hub (or your notary server of choice).
+
+> At this point, Connaisseur only supports BasicAuth for connecting to the notary.
+
+Below is a list of commands to programmatically do the changes to the `value.yaml` file in the `helm` folder. The openssl command will take a password as input, so you won't be able to copy+paste past it. However, you can also do these changes manually if you want. They are comprised of setting the notary server, notary credentials and the root publiy key for verification:
+
+```bash
+sed -i "s/host: notary.docker.io/host: ${NOTARY_URL}/" helm/values.yaml
+sed -i "s/user: notaryuser/user: ${NOTARY_USER}/" helm/values.yaml
+sed -i "s/password: Password123/password: ${NOTARY_PASSWORD}/" helm/values.yaml
+
+FIRST_LINE=$(grep -n rootPubKey: helm/values.yaml | sed s/:.*//)
+LAST_LINE=$(grep -nF -- '-----END PUBLIC KEY-----' helm/values.yaml | sed s/:.*//)
+sed -i $((${FIRST_LINE} + 1)),${LAST_LINE}d helm/values.yaml
+
+cd ~/.docker/trust/private
+sed '/^role:\sroot$/d' $(grep -iRl "role: root" .) > root-priv.key
+openssl ec -in root-priv.key -pubout -out root-pub.pem
+sed -i "${FIRST_LINE}s/.*/  rootPubKey: |\n    $(sed ':a;N;$!ba;s/\n/\\n    /g' root-pub.pem)/" ${OLDPWD}/helm/values.yaml
+rm root-priv.key root-pub.pem
+cd -
+```
```

Comment:
One could set Conny to detection mode which would make sure to not break anything, but might make seeing the outcome of using Conny less explicit (either warning or nothing).

Reply:
I wouldn't do that (by default). But since it is a nice feature, I would be inclined to mention it, now that you mention it (hehe). Like "hey if y'all wanna use non-blocking mode, you could also run "this command here", which is nice for checking at first but just doesnt give you as much sec, so y'all probably dont wanna do that in prod long-term.
That fine by you?

Reply:
Would go for smth like that. Like "this could be useful for testing or rolling out", but wont block the ima just give a log message/warning